packages/webapp/src/elections/components/participation-card.tsx

```diff
@@ -0,0 +1,287 @@
+            preventScroll
+            shouldCloseOnOverlayClick={!isLoading}
+            shouldCloseOnEsc={!isLoading}
+        >
+            <form onSubmit={onSubmit}>
+                <div className="space-y-4">
+                    <Text>
+                        You are committing to participate in the upcoming
+                        election. Not showing up could impact your standing and
+                        reputation in the community. If for some reason you
+                        cannot participate in the election, please update your
+                        status more than 24 hours from the election.
+                    </Text>
+                    <div className="p-3 border rounded">
+                        <Form.Checkbox
+                            id="confirm-checkbox"
+                            label="I agree to keep my participation status up to date."
+                            disabled={isLoading}
+                            required
+                        />
+                    </div>
+                    <div className="flex space-x-3">
+                        <Button
+                            type="neutral"
+                            onClick={close}
+                            disabled={isLoading}
+                        >
+                            Cancel
+                        </Button>
+                        <Button
+                            isSubmit
+                            isLoading={isLoading}
+                            disabled={isLoading}
+                        >
+                            {isLoading ? "Submitting..." : "Confirm"}
+                        </Button>
+                    </div>
+                </div>
+            </form>
+        </Modal>
+    );
+};
+
+const CancelParticipationModal = ({ isOpen, close }: ModalProps) => {
+    const [isLoading, setIsLoading] = useState<boolean>(false);
+    const [ualAccount] = useUALAccount();
+    const queryClient = useQueryClient();
+
+    const onSubmit = async (e: React.FormEvent) => {
+        e.preventDefault();
+
+        setIsLoading(true);
+
+        try {
+            const authorizerAccount = ualAccount.accountName;
+            const transaction = setElectionParticipation(
+                authorizerAccount,
+                false
+            );
+            console.info("signing trx", transaction);
+
+            const signedTrx = await ualAccount.signTransaction(transaction, {
+                broadcast: true,
+            });
+            console.info("electopt trx", signedTrx);
+
+            // invalidate current member query to update participating status
+            queryClient.invalidateQueries(
+                queryMemberByAccountName(ualAccount.accountName).queryKey
+            );
+
+            close();
+        } catch (error) {
+            console.error(error);
+            onError(error);
+        }
+
+        setIsLoading(false);
+    };
```

Comment:
Almost all of this `onSubmit` is like the one above it. Could be shared...although I realize this is a work in progress and you may be consciously duplicating them for now.

Reply:
Exactly that, because the other onsubmit will need to consider password confirmation!